src/tests/src/solver/optim-model-filler/test_componentFiller.cpp

```diff
@@ -738,4 +732,100 @@ BOOST_AUTO_TEST_CASE(one_var_with_param_objective)
+    buildLinearProblem();
+    BOOST_CHECK_EQUAL(pb->getObjectiveOffset(), 0);
+}
+
+// One var with offset
+BOOST_AUTO_TEST_CASE(one_var_with_offset)
+{
+    const auto objective = add(variable("x", 0), literal(10));
+    createModelWithOneFloatVar("model", {}, "x", literal(-50), literal(-40), {}, objective);
+    createComponent("model", "componentA", {});
+    buildLinearProblem();
+
+    BOOST_CHECK_EQUAL(pb->getObjectiveOffset(), 10);
+}
+
+BOOST_AUTO_TEST_CASE(one_param_offset)
+{
+    auto objective = parameter("param");
+    createModelWithOneFloatVar("model", {"param"}, "x", literal(-50), literal(-40), {}, objective);
+    createComponent("model", "componentA", {build_context_parameter_with("param", "5")});
+    buildLinearProblem();
+    BOOST_CHECK_EQUAL(pb->getObjectiveOffset(), 5);
+}
+
+BOOST_AUTO_TEST_CASE(one_time_dependent_var_with_offset)
+{
+    auto objective = add(variable("x", 0, TimeIndex::VARYING_IN_TIME_ONLY), literal(10));
+
+    createModelWithOneFloatVar("model", {}, "x", literal(-50), literal(-40), {}, objective, true);
+    createComponent("model", "componentA", {});
+
+    constexpr unsigned int last_time_step = 9;
+    FillContext ctx{0, last_time_step, 0, last_time_step, 0};
+    buildLinearProblem(ctx);
+    const auto nb_var = ctx.getLocalNumberOfTimeSteps(); // = 10
+
+    BOOST_CHECK_EQUAL(pb->variableCount(), nb_var);
+    for (unsigned i = 0; i < nb_var; i++)
+    {
+        const auto var_name = "componentA.x_s0_t" + to_string(i);
+        BOOST_CHECK_NO_THROW((void)pb->lookupVariable(var_name));
+        BOOST_CHECK_EQUAL(pb->getObjectiveOffset(), 10);
+    }
+}
+
+BOOST_AUTO_TEST_CASE(one_var_with_time_dependent_offset)
+{
+    auto objective = add(variable("x", 0), parameter("param", TimeIndex::VARYING_IN_TIME_ONLY));
+    createModelWithSystemModelParameter(
+      "model",
+      {Parameter{"param", TimeDependent::YES, ScenarioDependent::NO}},
+      {{"x", ValueType::FLOAT, literal(-5), literal(10), true, false}},
+      {},
+      objective);
+    createComponent("model",
+                    "componentA",
+                    {build_context_parameter_with("param", "bounds", ParameterType::TIMESERIE)});
+
+    const vector<unsigned int> timeSteps{0, 1, 2};
+    FillContext ctx{timeSteps.at(0), timeSteps.at(2), timeSteps.at(0), timeSteps.at(2), 0};
+    auto bounds_time_series = std::make_unique<TimeSeriesSet>("bounds", 3);
+    // setting 3 hours (including h 1 and 2)
+    bounds_time_series->add({10., 11., 12.});
+    LinearProblemData data;
+    data.addDataSeries(std::move(bounds_time_series));
+
+    std::vector<std::unique_ptr<IScenario>> scenarios;
+    try
+    {
+        buildLinearProblem(ctx, data, scenarios);
+    }
+    catch (const Antares::Error::RuntimeError& e)
+    {
+        BOOST_CHECK(std::string(e.what()).find(
+                      "Trying to set multiple objective offset for the same objective.")
+                    != std::string::npos);
+    }
```

Comment:
Instead of this **try**... **catch**, we already have : 
```c++
BOOST_CHECK_EXCEPTION(buildLinearProblem(ctx, data, scenarios),
                      std::runtime_error,
                      containsMessage(err_msg));
```